samples/boards/nrf/nrfx_prs/src/main.c

```diff
@@ -0,0 +1,396 @@
+		printk("nrfx_spim_xfer() failed: 0x%08x\n", err);
+		return false;
+	}
+
+	if (k_sem_take(&transfer_finished, K_MSEC(100)) != 0) {
+		printk("SPIM transfer timeout\n");
+		return false;
+	}
+
+	received = rx_buf_size;
+	return true;
+}
+
+static void uarte_handler(const nrfx_uarte_event_t *p_event, void *p_context)
+{
+	if (p_event->type == NRFX_UARTE_EVT_RX_DONE) {
+		received = p_event->data.rxtx.bytes;
+		k_sem_give(&transfer_finished);
+	} else if (p_event->type == NRFX_UARTE_EVT_ERROR) {
+		received = 0;
+		k_sem_give(&transfer_finished);
+	}
+}
+
+static bool switch_to_uarte(void)
+{
+	if (uarte_initialized) {
+		return true;
+	}
+
+	/* If the SPIM is currently initialized, it must be deinitialized
+	 * before the UARTE can be used.
+	 */
+	if (spim_initialized) {
+		nrfx_spim_uninit(&spim);
+		spim_initialized = false;
+	}
+
+	nrfx_err_t err;
+	nrfx_uarte_config_t uarte_config = NRFX_UARTE_DEFAULT_CONFIG(
+		/* Take pin numbers from devicetree. */
+		DT_PROP(UARTE_NODE, tx_pin),
+		DT_PROP(UARTE_NODE, rx_pin));
+	uarte_config.baudrate = NRF_UARTE_BAUDRATE_1000000;
+
+	err = nrfx_uarte_init(&uarte, &uarte_config, uarte_handler);
+	if (err != NRFX_SUCCESS) {
+		printk("nrfx_uarte_init() failed: 0x%08x\n", err);
+		return false;
+	}
+
+	uarte_initialized = true;
+	printk("Switched to UARTE\n");
+	return true;
+}
+
+static bool uarte_transfer(const uint8_t *tx_data, size_t tx_data_len,
+			   uint8_t *rx_buf, size_t rx_buf_size)
+{
+	nrfx_err_t err;
+
+	err = nrfx_uarte_rx(&uarte, rx_buf, rx_buf_size);
+	if (err != NRFX_SUCCESS) {
+		printk("nrfx_uarte_rx() failed: 0x%08x\n", err);
+		return false;
+	}
+
+	err = nrfx_uarte_tx(&uarte, tx_data, tx_data_len);
+	if (err != NRFX_SUCCESS) {
+		printk("nrfx_uarte_tx() failed: 0x%08x\n", err);
+		return false;
+	}
+
+	if (k_sem_take(&transfer_finished, K_MSEC(100)) != 0) {
+		/* The UARTE transfer finishes when the RX buffer is completely
+		 * filled. In case the UARTE receives less data (or nothing at
+		 * all) within the specified time, taking the semaphore will
+		 * fail. In such case, stop the reception and end the transfer
+		 * this way. Now taking the semaphore should be successful.
+		 */
+		nrfx_uarte_rx_abort(&uarte);
+		if (k_sem_take(&transfer_finished, K_MSEC(10)) != 0) {
+			printk("UARTE transfer timeout\n");
+			return false;
+		}
+	}
+
+	return true;
+}
+
+static void buffer_dump(const uint8_t *buffer, size_t length)
+{
+	for (int i = 0; i < length; ++i) {
+		printk(" %02X", buffer[i]);
+	}
+	printk("\n");
+}
+
+static bool background_transfer(const struct device *spi_dev)
+{
+	static const uint8_t tx_buffer[] = "Nordic Semiconductor";
+	static uint8_t rx_buffer[sizeof(tx_buffer)];
+	static const struct spi_cs_control spi_dev_cs_ctrl = {
+		.gpio_dev = DEVICE_DT_GET(DT_GPIO_CTLR(SPI_DEV_NODE, cs_gpios)),
+		.gpio_pin = DT_GPIO_PIN(SPI_DEV_NODE, cs_gpios),
+		.gpio_dt_flags = DT_GPIO_FLAGS(SPI_DEV_NODE, cs_gpios)
+	};
+	static const struct spi_config spi_dev_cfg = {
```

Comment:
I think you can get away with using SPI_CONFIG_DT here instead.

Reply:
This SPI controller does not have any child nodes defined and I would need one to pass its node identifier as the parameter for `SPI_CONFIG_DT`. I'm not sure it is worth adding a child node just to be able to use this macro. The intent here is only to show that it is possible to use the Zephyr SPI driver for one peripheral instance while using the nrfx SPIM driver for another one.

Reply:
OK, sounds like a good reason not to use that.